### PR TITLE
feat: Birthdays

### PR DESCRIPTION
**Pull request by Hypeism**
feat: Birthdays

## Discord Username
HowNowBrownCrow#2591

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,4 @@ node_modules
 .idea/
 *.iml
 build/
+
```
Comment (NathanInnes) [inline .gitignore:6]: Why extra line?

File: src/commands/utils/birthdayremove [hunk 2]
```diff
@@ -0,0 +1,69 @@
+import { CommandDefinition } from '../../lib/command';
```
Comment (NathanInnes) [inline src/commands/utils/birthdayremove:1]: This file has no extension type.

File: birthdays.session.sql [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+-- @block
+SELECT 
+    Users.id AS discord_id
+    birthday.id AS birthday_id,
+    discord_id
+    birthday
+
+FROM users
+INNER JOIN birthday ON birthday.id = Users.id;
```
[the remark this answers is not in the document]
Comment (benw202) [reply birthdays.session.sql:34]: I believe this is a basic first look inspired by - https://www.npmjs.com/package/discord-birthdays

Will need a whole re-work

File: .gitignore (hunk 1 above)
Comment (Hypeism) [reply .gitignore]: Accidental, this is why I should probably not try programming at 3am lol. Just undid the change with a new commit

File: birthdays.session.sql (hunk 3 above)
[the remark this answers is not in the document]
Comment (Hypeism) [reply birthdays.session.sql:34]: > Curious as to where this file was copied from?

the sql was setup practice from https://www.youtube.com/watch?v=Cz3WcZLRaWc&t=920s I am working on refactoring for birthday use

[the remark this answers is not in the document]
Comment (NathanInnes) [reply birthdays.session.sql:34]: All good was just making sure it conforms to any sort of license agreement of the creator 🙂

File: src/commands/utils/birthdayremove (hunk 2 above)
Comment (Hypeism) [reply src/commands/utils/birthdayremove]: fixed, thank you

File: src/handlers/birthdayhandler.ts [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+                        }
+                    }
+                });
+            }
+        });
+    }, 120000) // Every  46.8 minmutes (avoids double pings)
+}
+
+export async function addbday (con, userid, date) {
+    await con.query(`SELECT * FROM birthdays WHERE userid='${userid}'`, async (err, row) => {
```
Comment (Benjozork) [inline src/handlers/birthdayhandler.ts:75]: This **CANNOT** make it to production! This is a risk of SQL injection. Please read up on this:

https://www.npmjs.com/package/mysql#escaping-query-values

This should be applied to every query.

**Review by Benjozork — CHANGES_REQUESTED** [review]

**Pull request closed without merging**